cmd/proxy/actions/basicauth.go

```diff
@@ -14,26 +14,27 @@ const healthWarning = "/healthz received none or incorrect Basic-Auth headers"
 func basicAuth(user, pass string) mux.MiddlewareFunc {
 	return func(h http.Handler) http.Handler {
 		f := func(w http.ResponseWriter, r *http.Request) {
-			if !checkAuth(r, user, pass) {
+			if isHealthEndpoint(r) {
```

Comment:
@manugupt1 
What do you think about passing a slice of routes (maybe `mux.Route`s or just paths?) that should be excluded to the middleware?
Currently it's impossible to know that the health route will be excluded without looking at the implementation.

Reply:
My vote is for passing the slice of routes because it's less information but still gets the job done

Reply:
Yes! I am trying to figure out a way to separate out the handlers in a way so that we can add more middleware if necessary to the new handlers.